lib/components/ClinicUserEdit.js

```diff
@@ -233,40 +150,78 @@ var ClinicUserEdit = React.createClass({
             {_.get(this.props.allUsers, [this.props.loggedInUser, 'fullName'])}
           </div>
         </div>
-        <form className={styles.form}>
+        <form className={styles.form} onSubmit={handleSubmit(this.handleNext)}>
           <div className={styles.inputWrap}>
-            <label className={styles.inputLabel} htmlFor="name">Patient Full Name</label>
-            <input className={styles.input} ref="name" name='name' value={this.state.name} onChange={this.handleChange} onBlur={this.handleBlur} placeholder=""/>
+            <label className={styles.inputLabel} htmlFor="name">
+              Patient Full Name
+            </label>
+            <input className={styles.input} {...fullName}/>
+            {fullName.touched && fullName.error && <div className={styles.validationError}>{fullName.error}</div>}
```

Comment:
Logical && will return the rightmost value if all values are truthy. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_AND_() (line 5). 
I think you're right about `touched`, I'll revise it for clarity.